src/actions/WalletActions.js

```diff
@@ -143,6 +144,12 @@ export const selectEOSWallet = (walletId: string, currencyCode: string, from?: s
   const guiWallet = UI_SELECTORS.getWallet(state, walletId)
   if (walletId !== currentWalletId || currencyCode !== currentWalletCurrencyCode || from === Constants.WALLET_LIST_SCENE) {
     const { publicAddress } = guiWallet.receiveAddress
+    if (!publicAddress) {
+      // Update all wallets' addresses. Hopefully gets the updated address for the next time
+      // We enter the EOS wallet
+      dispatch(updateWalletsRequest())
+    }
+
     if (publicAddress) {
       // already activated
```

Comment:
I just want to point out that there is already an `else` clause for the existence of `publicAddress`